Packet++/src/LdapLayer.cpp

```diff
@@ -214,6 +330,77 @@ namespace pcpp {
 	}
 	// endregion
 
+	// region LdapResponseLayer
+
+	LdapResponseLayer::LdapResponseLayer(uint16_t messageId, const LdapOperationType& operationType, const LdapResultCode& resultCode,
+		const std::string& matchedDN, const std::string& diagnosticMessage, const std::vector<std::string>& referral,
+		const std::vector<LdapControl>& controls)
+	{
+		Asn1EnumeratedRecord resultCodeRecord(resultCode);
+		Asn1OctetStringRecord matchedDNRecord(matchedDN);
+		Asn1OctetStringRecord diagnosticMessageRecord(diagnosticMessage);
+
+		std::vector<Asn1Record*> messageRecords = {&resultCodeRecord, &matchedDNRecord, &diagnosticMessageRecord};
+
+		std::unique_ptr<Asn1ConstructedRecord> referralRecord;
+		if (!referral.empty())
+		{
+			PointerVector<Asn1Record> referralSubRecords;
+			for (const auto& uri : referral)
+			{
+				referralSubRecords.pushBack(new Asn1OctetStringRecord(uri));
+			}
+			referralRecord = std::unique_ptr<Asn1ConstructedRecord>(new Asn1ConstructedRecord(
+				Asn1TagClass::ContextSpecific, referralTagType, referralSubRecords));
+			messageRecords.push_back(referralRecord.get());
+		}
+
+		LdapLayer::init(messageId, operationType, messageRecords, controls);
+	}
+
+	LdapResultCode LdapResponseLayer::getResultCode() const
+	{
+		return LdapResultCode::fromUintValue(getLdapOperationAsn1Record()->getSubRecords().at(resultCodeIndex)->castAs<Asn1EnumeratedRecord>()->getValue());
+	}
+
+	std::string LdapResponseLayer::getMatchedDN() const
+	{
+		return getLdapOperationAsn1Record()->getSubRecords().at(matchedDNIndex)->castAs<Asn1OctetStringRecord>()->getValue();
+	}
+
+	std::string LdapResponseLayer::getDiagnosticMessage() const
+	{
+		return getLdapOperationAsn1Record()->getSubRecords().at(diagnotsticsMessageIndex)->castAs<Asn1OctetStringRecord>()->getValue();
+	}
+
+	std::vector<std::string> LdapResponseLayer::getReferral() const
+	{
+		std::vector<std::string> result;
+		if (getLdapOperationAsn1Record()->getSubRecords().size() <= referralIndex)
+		{
+			return result;
+		}
+
+		auto referralRecord = getLdapOperationAsn1Record()->getSubRecords().at(referralIndex);
+		if (referralRecord->getTagClass() != Asn1TagClass::ContextSpecific || referralRecord->getTagType() != referralTagType)
+		{
+			return result;
+		}
+
+		for (auto uriRecord : referralRecord->castAs<Asn1ConstructedRecord>()->getSubRecords())
+		{
+			result.push_back(uriRecord->castAs<Asn1OctetStringRecord>()->getValue());
+		}
+
+		return result;
+	}
+
+	std::string LdapResponseLayer::getExtendedStringInfo() const
```

Comment:
maybe rename to `getExtendedInfoString`?

Reply:
I agree, updated in this commit: https://github.com/seladb/PcapPlusPlus/pull/1437/commits/c72a2af0fc7b9c3f0dbb9b03f4cf77dc75d5bc9d